sdk/core/Azure.Core/samples/Diagnostics.md

```diff
@@ -72,23 +79,84 @@ SecretClientOptions options = new SecretClientOptions()
-
-
 ### Custom logging callback
 
 The `AzureEventSourceListener` class can also be used with a custom callback that allows log messages to be written to destination of your choice.
 
 ```C# Snippet:LoggingCallback
 using AzureEventSourceListener listener = new AzureEventSourceListener(
-    (e, message) => Console.WriteLine("[{0:HH:mm:ss:fff}][{1}] {2}", DateTimeOffset.Now, e.Level, message),
+    (args, message) => Console.WriteLine("[{0:HH:mm:ss:fff}][{1}] {2}", DateTimeOffset.Now, args.Level, message),
     level: EventLevel.Verbose);
 ```
 
-When targeting .NET Standard 2.1, .NET Core 2.2, or newer, you might instead use `e.TimeStamp` to log the time the event was written instead of rendered, like above. It's in UTC format, so if you want to log the local time like in the example call `ToLocaleTime()` first.
-For help diagnosing multi-threading issues, you might also log `e.OSThreadId` which is also available on those same targets.
+When targeting .NET Standard 2.1, .NET Core 2.2, or newer, you might instead use `args.TimeStamp` to log the time the event was written instead of rendered, like above. It's in UTC format, so if you want to log the local time like in the example call `ToLocaleTime()` first.
+For help diagnosing multi-threading issues, you might also log `args.OSThreadId` which is also available on those same targets.
+
+More information about the `args` parameter for the callback can be found in the [EventWrittenEventArgs](https://docs.microsoft.com/dotnet/api/system.diagnostics.tracing.eventwritteneventargs) documentation.
+
+### Applying filtering logic
+
+The custom callback can be used with the listener to help filter log messages to reduce volume and noise when troubleshooting.   
+
+In the following example, `Verbose` messages for the `Azure-Identity` event source are captured and written to `Trace`.  Log messages for the `Azure-Messaging-EventHubs` event source are filtered to capture only a specific set to aid in debugging publishing, which are then written to the console.
+
+```C# Snippet:LoggingWithFilters
+using AzureEventSourceListener listener = new AzureEventSourceListener((args, message) =>
+{
+    if (args.EventSource.Name.StartsWith("Azure-Identity") && args.Level == EventLevel.Verbose)
+    {
+        Trace.WriteLine(message);
+    }
+    else if (args.EventSource.Name.StartsWith("Azure-Messaging-EventHubs"))
+    {
+        switch (args.EventId)
+        {
+            case 3:   // Event Publish Start
+            case 4:   // Event Publish Complete
+            case 5:   // Event Publish Error
+                Console.WriteLine(message);
+                break;
+        }
+    }
+}, EventLevel.LogAlways);
+```
+
+### Capture filtered logs to a file
```

Comment:
Logging to a file or database is one of the most common questions that I've seen asked.